torch/_ops.py

```diff
@@ -588,9 +698,45 @@ def handler(*args, **kwargs):
             add_cached_op(self)
             return handler
 
-        cache_result = True
         functionality_key = torch._C._to_functionality_key(key)  # type: ignore[attr-defined]
-        if functionality_key in mode_stack_per_key():
+        if functionality_key == torch._C.DispatchKey.PreDispatch:
```

Comment:
this logic looks like it's mostly a duplicate of the `elif functionality_key in mode_stack_per_key():` below it. I'd rather avoid the duplication, can we just assert / raise an error on the second case, or delete per-dispatch-key-modes completely?